src/Microsoft.TestPlatform.CrossPlatEngine/EventHandlers/TestRequestHandler.cs

```diff
@@ -141,14 +158,16 @@ public void Close()
     /// <inheritdoc />
     public void SendTestCases(IEnumerable<TestCase> discoveredTestCases)
     {
-        var data = _dataSerializer.SerializePayload(MessageType.TestCasesFound, discoveredTestCases, _protocolVersion);
+        var updatedTestCases = _pathConverter.UpdateTestCases(discoveredTestCases, Direction.Send);
```

Comment:
Every incoming and outgoing requests is modified and paths are changed based on the request direction, when we Send we replace deployDirectory path with the directory into which we built on the original system, when we recieve the request we do it the other way around. 

```
Receive C:\t\mstest1\bin\net6.0\mstest1.dll -> C:\deploy\mstest1.dll
Send    C:\deploy\mstest1.dll -> C:\t\mstest1\bin\net6.0\mstest1.dll
```